add urdf generated with cad and onshape to robot

pierfabre opened the pull request:
closes #93 

Add an urdf  of the robot (v2) generated with onshape-to-robot

pierfabre committed:
add urdf generated with cad and onshape to robot

Armandpl commented:
- why do we need new stls? Can't we use those already in the repo?
- we also want to migrate to git lfs for CAD files. now might be a good time? We want to rewrite the git history/use git lfs migrate to make sure we remove current files from the history

Armandpl commented:
maybe we also want a urdf for both versions of the robot until we merge it into a single robot? which we should probably do soon?

pierfabre commented:
> * why do we need new stls? Can't we use those already in the repo?  

most of them were not on the repo

> * we also want to migrate to git lfs for CAD files. now might be a good time? We want to rewrite the git history/use git lfs migrate to make sure we remove current files from the history  

do you want to do it?

> maybe we also want a urdf for both versions of the robot until we merge it into a single robot? which we should probably do soon?  

I think we can stay with this version until we converge on one single version as (as far as I know) you don't need the urdf of your version for now. It's a bit of work to generate it and if you don't really need it I would prefer not to do it :)

pierfabre committed:
organize stl by v1/v2/clamp_assembly

pierfabre committed:
adapt names in urdf

pierfabre committed:
make Viewer3D and new simulation available

pierfabre committed:
merge STLs

pierfabre commented:
@Armandpl I simplified the urdf by merging stls from same link together. I also deleted the stls I had created that where unused. So in the end I add only 3 new stls to the repo (to visualize the 3 links), the rest is just reorganizing stls by v1/v2/clamp.

Armandpl commented:
the reason you need extra stls is to include stuff we don't print e.g the encoder? 
what are the steps to go from this to using the printable stls + extra stls for non printable parts instead of the "merged" stls? good to merge if it's very clear how we'll clean this up.

pierfabre committed:
simplify STLs

pierfabre commented:
> the reason you need extra stls is to include stuff we don't print e.g the encoder?

Yes exactly, and rather than adding all the missing parts' STLs individually I just add 3 merged STLs for the 3 body link

 > this is for visualization only and the sim could work without it? 

Yes

> what are the steps to go from this to using the printable stls + extra stls for non printable parts instead of the "merged" stls? 

It is feasible, my concern is that it will make the urdf very long and messy. Visualization may also be faster with only 3 STLs to display. Notice that I ran a pass of simplification in my last commit to make them a bit lighter.

> good to merge if it's very clear how we'll clean this up.

It will be cleaner the day we have only one design of course. We will have a folder for the printable parts' STLs but I will still need my 3 merged STLs for visualization.
